assembly/src/tokens/stream.rs

```diff
@@ -93,10 +114,54 @@ impl<'a> TokenStream<'a> {
             }
         }
     }
+
+    pub fn take_doc_comment_at(&mut self, pos: usize) -> Option<String> {
+        self.doc_comments.remove(&pos)?
+    }
 }
 
 impl<'a> fmt::Display for TokenStream<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{:?}", &self.tokens[self.pos..])
     }
 }
+
+#[derive(Debug)]
+pub struct Comment(Option<String>);
+
+impl Comment {
+    pub fn append_line(&mut self, line: &str) {
+        let prepared_line = Self::prepare_line(line);
+        if !prepared_line.is_empty() {
+            match &mut self.0 {
+                Some(comment) => {
+                    comment.push('\n');
+                    comment.push_str(prepared_line);
+                }
+                None => {
+                    let mut comment = String::new();
+                    comment.push_str(prepared_line);
+                    self.0 = Some(comment);
+                }
+            }
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.0.is_none()
+    }
+
+    pub fn take_content(&mut self) -> Option<String> {
+        self.0.take()
+    }
+
+    /// Removes `prefix` from provided `line` and trims additional whitespaces from start and end of
+    /// the `line`
+    fn prepare_line(line: &str) -> &str {
```

Comment:
nit: since this is just a function (not really related to `Comment` - I'd move out outside of `Comment` struct.